### PR TITLE
proto: improve comments

### PR DESCRIPTION
**Pull request by AkihiroSuda**
Signed-off-by: Akihiro Suda <suda.akihiro@lab.ntt.co.jp>

### CODE REVIEW

File: proto/manifest.proto [hunk 1]
```diff
@@ -12,22 +12,26 @@ message Resource {
     // Path specifies the path from the bundle root. If more than one
     // path is present, the entry may represent a hardlink, rather than using
     // a link target. The path format is operating system specific.
+    // Path MAY be an absolute path, but it MUST be interpreted with the bundle root.
```
Comment (AkihiroSuda) [inline proto/manifest.proto]: current `manifest_test.go` uses relative path but `continuity build` uses absolute path.
Suggestion: can we prohibit absolute path for ease of robust implementation?

File: proto/manifest.proto [hunk 2]
```diff
@@ -12,22 +12,26 @@ message Resource {
...
     // compatibility across different OS.
+    // On Unix, uid MUST be a positive decimal integer or an empty string.
     string uid = 2;
 
     // Gid specifies the group id for the resource. A string type is used for
     // compatibility across different OS.
+    // On Unix, gid MUST be a positive decimal integer or an empty string.
     string gid = 3;
 
+    // User and group specify the user name and the group name for the resource.
```
Comment (stevvooe) [inline proto/manifest.proto]: These fields aren't used, so this definition is meaningless.

File: proto/manifest.proto [hunk 3]
```diff
@@ -12,22 +12,26 @@ message Resource {
...
     // a link target. The path format is operating system specific.
+    // Path MAY be an absolute path, but it MUST be interpreted with the bundle root.
     repeated string path = 1;
 
     // NOTE(stevvooe): Need to define clear precedence for user/group/uid/gid precedence.
 
     // Uid specifies the user id for the resource. A string type is used for
     // compatibility across different OS.
+    // On Unix, uid MUST be a positive decimal integer or an empty string.
     string uid = 2;
```
Comment (stevvooe) [inline proto/manifest.proto]: Let's just change these to integers. Windows aren't gonna use these, so they'll always be integers. Use `int64`.

Comment (AkihiroSuda) [inline proto/manifest.proto]: * not uint64?
* please elaborate your plan for windows?

Comment (stevvooe) [inline proto/manifest.proto]: > please elaborate your plan for windows?

You'll have to ask the microsoft folks, but they made the indication that we are going to use separate fields for that purpose. The only reason these are strings is due to the fact Windows has string user ids.

> not uint64?

Don't use integer types to enforce signedness.

Comment (AkihiroSuda) [inline proto/manifest.proto]: cc @jhowardmsft

Comment (jhowardmsft) [inline proto/manifest.proto]: @AkihiroSuda https://github.com/stevvooe/continuity/pull/65#discussion_r113073051

Comment (AkihiroSuda) [inline proto/manifest.proto]: thank you, will update soon

**Comment by AkihiroSuda** [comment]
updated PR and opened https://github.com/containerd/continuity/pull/77 for unstringifying uid&gid

File: proto/manifest.proto (hunk 1 above)
Comment (AkihiroSuda) [inline proto/manifest.proto]: @stevvooe WDYT?

Comment (stevvooe) [inline proto/manifest.proto]: Do we need to have a change to make this work, as well?

Comment (stevvooe) [inline proto/manifest.proto]: We should always be using relative paths for continuity.

Comment (AkihiroSuda) [inline proto/manifest.proto]: except symlink, right? (e.g. some executable can be a symlink to `/bin/true` or `/bin/false` on the absolute root)

**Comment by stevvooe** [comment]
@AkihiroSuda Paths should be relative.

**Comment by AkihiroSuda** [comment]
ok, I will open a PR in a few days

**Comment by AkihiroSuda** [comment]
updated PR, and opened #82

**Comment by stevvooe** [comment]
LGTM

**Pull request merged**